main.go

```diff
@@ -0,0 +1,275 @@
+		s.Configuration = &Configuration{
+			URLs: make(map[string]configurationtypes.URL),
+		}
+	}
+
+	if app.DefaultCache == nil {
+		return nil
+	}
+
+	if s.Configuration.DefaultCache == nil {
+		s.Configuration.DefaultCache = &DefaultCache{
+			Headers: app.Headers,
+			TTL:     app.TTL,
+		}
+	} else {
+		dc := s.Configuration.DefaultCache
+		appDc := app.DefaultCache
+		if dc.Headers == nil {
+			s.Configuration.DefaultCache.Headers = appDc.Headers
+		}
+		if dc.TTL == "" {
+			s.Configuration.DefaultCache.TTL = appDc.TTL
+		}
+		if dc.Olric.URL == "" || dc.Olric.Path == "" || dc.Olric.Configuration == nil {
+			s.Configuration.DefaultCache.Distributed = appDc.Distributed
+			s.Configuration.DefaultCache.Olric = appDc.Olric
+		}
+	}
+
+	return nil
+}
+
+// Provision to do the provisioning part.
+func (s *SouinCaddyPlugin) Provision(ctx caddy.Context) error {
+	s.logger = ctx.Logger(s)
+
+	if err := s.configurationPropertyMapper(); err != nil {
+		return err
+	}
+
+	app, _ := ctx.App(moduleName)
+
+	if err := s.FromApp(app.(*SouinApp)); err != nil {
+		return err
+	}
+
+	s.bufPool = &sync.Pool{
+		New: func() interface{} {
+			return new(bytes.Buffer)
+		},
+	}
+	s.Retriever = plugins.DefaultSouinPluginInitializerFromConfiguration(s.Configuration)
+	s.RequestCoalescing = coalescing.Initialize()
+	return nil
+}
+
+func parseCaddyfileRecursively(h *caddyfile.Dispenser) interface{} {
+	input := make(map[string]interface{})
+	for nesting := h.Nesting(); h.NextBlock(nesting); {
+		val := h.Val()
+		if val == "}" || val == "{" {
+			continue
+		}
+		args := h.RemainingArgs()
+		if len(args) > 0 {
+			input[val] = args[0]
+		} else {
+			input[val] = parseCaddyfileRecursively(h)
+		}
+	}
+
+	return input
+}
+
+func parseCaddyfileGlobalOption(h *caddyfile.Dispenser, _ interface{}) (interface{}, error) {
+	souinApp := new(SouinApp)
+	cfg := &Configuration{
+		DefaultCache: &DefaultCache{
+			Distributed: false,
+			Headers:     []string{},
+		},
+		URLs: make(map[string]configurationtypes.URL),
+	}
+
+	for h.Next() {
+		for nesting := h.Nesting(); h.NextBlock(nesting); {
+			rootOption := h.Val()
+			switch rootOption {
+			case "headers":
+				args := h.RemainingArgs()
+				cfg.DefaultCache.Headers = append(cfg.DefaultCache.Headers, args...)
+			case "log_level":
+				args := h.RemainingArgs()
+				cfg.LogLevel = args[0]
+			case "olric":
+				cfg.DefaultCache.Distributed = true
+				provider := configurationtypes.CacheProvider{}
+				for nesting := h.Nesting(); h.NextBlock(nesting); {
+					directive := h.Val()
+					switch directive {
+					case "url":
+						urlArgs := h.RemainingArgs()
+						provider.URL = urlArgs[0]
+					case "path":
+						urlArgs := h.RemainingArgs()
+						provider.Path = urlArgs[0]
+					case "configuration":
+						provider.Configuration = parseCaddyfileRecursively(h)
+					}
+				}
+				cfg.DefaultCache.Olric = provider
+			case "ttl":
+				args := h.RemainingArgs()
+				cfg.DefaultCache.TTL = args[0]
```

Comment:
You should use `caddy.ParseDuration()` here.

Reply:
Related to the core: https://github.com/Darkweak/Souin/issues/94